### ISSUE TITLE
Unable to view predictions for custom dataset

### ISSUE THREAD

**Issue by EricWiener** [ultralytics/yolov5#586]
## ❔Question
I attempted to train Yolov5 using a custom dataset based on the provided Colab Notebook, but I am unable to get the test predictions to show up. The bounding boxes are correctly showing up for the training and validation data, so I believe the images are labeled in the correct format. The Google Drive folder with all relevant information is [here](https://drive.google.com/drive/folders/1DQ5DM0ZheZHqsdIwz8RDDWzw3Y0JHiuO?usp=sharing). In the shared notebook, the model is trained for around 300 epochs currently, but I have previously attempted with 1000 epochs and I receive the same results every time.

Additionally, when attempting to plot the results in the Colab Notebook, I get the message `No handles with labels found to put in legend`. However, when I look at `/content/yolov5/runs/exp1/results.png`, I can see the results plotted as expected. 

Thank you for your time and for open sourcing your work. I would really appreciate any help/advice you can provide.

Thanks,
Eric

**Comment by glenn-jocher** [comment]
@EricWiener I would simply follow the tutorial example, train coco128 for 300 epochs, then use last.pt with detect.py. Once you have this working, train on your own data instead.

If you've done this and detect.py is not producing detections on new images, then your data or your training are not capable of generalizing to unseen images. If detect.py is not producing detections on validation images while test.py is, then you're doing something incorrectly.

**Comment by EricWiener** [comment]
@glenn-jocher Thanks for the response! I got it working for the coco128 and then tried it with my custom data, but it stopped working. Could it be an issue that not every class is represented in the training data (I planned on using all classes later)? I'll try using all classes and will update if that helps

**Comment by glenn-jocher** [comment]
@EricWiener great! This is why we provide easy examples, to help you isolate the source of problems. If the training pipeline operates correctly on the default data, and then stops working when you modify that one piece, then you know the source of the problem.

**Comment by EricWiener** [comment]
Updating to the newest version of the repo fixed the problem. It is now working correctly.
